File: market-app/src/main/java/com/seminav/marketapp/services/UserService.java
package com.seminav.marketapp.services;

import com.seminav.marketapp.external.messages.UserDto;
import com.seminav.marketapp.messages.GetMyProductsResponse;
import com.seminav.marketapp.model.User;

public interface UserService {
    User getUserOrElseSave(UserDto userDto);

    GetMyProductsResponse getProducts(String userId);

    User save(User user);
}

File: user-app/src/main/kotlin/com/missclick3/services/user/UserService.kt
package com.missclick3.services.user

import com.missclick3.messages.dtos.UserDTO
import com.missclick3.messages.requests.PatchUserByAdminRequest
import com.missclick3.messages.requests.PatchUserPersonalInfoRequest
import com.missclick3.model.User
import java.util.UUID

interface UserService {
    suspend fun createNewUser(userDTO: UserDTO): Boolean

    suspend fun getUserById(id: UUID): UserDTO?

    suspend fun getUserByUsername(username: String): UserDTO?

    suspend fun updateUserPersonalInfo(id: UUID, request: PatchUserPersonalInfoRequest): Boolean

    suspend fun updateUser(id: UUID, request: PatchUserByAdminRequest): Boolean

    suspend fun deleteUser(id: UUID): Boolean

    suspend fun getUsers(): List<UserDTO>
}